cosmwasm/enclaves/shared/block-verifier/src/submit_block_signatures.rs

```diff
@@ -0,0 +1,207 @@
+use std::slice;
+
+use tendermint_proto::Protobuf;
+
+use sgx_types::sgx_status_t;
+
+use enclave_utils::{validate_const_ptr, validate_input_length, validate_mut_ptr};
+use log::error;
+
+use log::debug;
+
+use tendermint::validator::Set;
+
+macro_rules! unwrap_or_error {
+    ($result:expr) => {
+        match $result {
+            Ok(commit) => commit,
+            Err(e) => return e.into(),
+        }
+    };
+}
+
+use crate::txs::tx_from_bytes;
+use crate::wasm_messages::VERIFIED_MESSAGES;
+
+use crate::verify::validator_set::get_validator_set_for_height;
+use enclave_utils::validator_set::ValidatorSetForHeight;
+
+const MAX_VARIABLE_LENGTH: u32 = 100_000;
+const RANDOM_PROOF_LEN: u32 = 80;
+const MAX_TXS_LENGTH: u32 = 10 * 1024 * 1024;
+const TX_THRESHOLD: usize = 100_000;
+
+#[no_mangle]
+#[allow(unused_variables)]
+pub unsafe fn submit_block_signatures_impl(
+    in_header: *const u8,
+    in_header_len: u32,
+    in_commit: *const u8,
+    in_commit_len: u32,
+    in_txs: *const u8,
+    in_txs_len: u32,
+    in_encrypted_random: *const u8,
+    in_encrypted_random_len: u32,
+    decrypted_random: &mut [u8; 32],
+) -> sgx_status_t {
+    if let Err(e) = validate_inputs(
+        in_header,
+        in_header_len,
+        in_commit,
+        in_commit_len,
+        in_txs,
+        in_txs_len,
+        in_encrypted_random,
+        in_encrypted_random_len,
+        decrypted_random,
+    ) {
+        return e;
+    }
+
+    let block_header_slice = slice::from_raw_parts(in_header, in_header_len as usize);
+    let block_commit_slice = slice::from_raw_parts(in_commit, in_commit_len as usize);
+
+    let txs_slice = if in_txs_len != 0 && !in_txs.is_null() {
```

Comment:
From what I've checked in rust playground when in_txs is null or in_txs_len is 0 you always get an empty array so I think the if is redundant here you can change it to 
```suggestion
    let txs_slice = slice::from_raw_parts(in_txs, in_txs_len as usize)
```

Reply:
I added this here because this caused a crash when txs was empty. I'll leave it like this and we test it a bit more later on